lib/rust/code-builder/src/lib.rs

```diff
@@ -56,13 +56,17 @@ impl CodeBuilder {
     /// Create a new line and insert appropriate indentation.
     pub fn newline(&mut self) {
         let space_count = self.spaces_in_indent * self.indent;
-        self.write(format!("\n{}", " ".repeat(space_count)));
+        self.buffer.reserve(space_count + 1);
+        self.buffer.push('\n');
+        for _ in 0..space_count {
+            self.buffer.push(' ');
+        }
```

Comment:
Why this change? The new code is WAY harder to read than the old one.

Reply:
The previous implementation allocated two separate temporary strings. The new one has zero temporary allocs and at most once reallocated the string to grow it.

Reply:
I know, but ugh, its ugly. But yeah, lets keep the performance maxed out.